animated-base/src/main/java/com/facebook/fresco/animation/bitmap/cache/FrescoFrameCache.java

```diff
@@ -29,25 +29,47 @@
 public class FrescoFrameCache implements BitmapFrameCache {
 
   private static final Class<?> TAG = FrescoFrameCache.class;
+  private static final int NO_VALUE = -1;
 
   private final AnimatedFrameCache mAnimatedFrameCache;
+  private AnimatedFrameCache mOtherFrameCache;
```

Comment:
When we scroll in a webP list, if we put all webP frames in one FrameCache, the outwindow webP is easier to be recycled. I used a otherFrameCahce to cache other frames and mAnimatedFrameCache to cache first frame. Then when we scroll in a webP list, the other frames can be recycled, the first frame remains, it can take less time for users to preview the image.

Reply:
If you always want to keep the first frame (which should be an option in ImageDecodeOptions), I suggest to just hold a reference to the bitmap in the backend and close it when the backend is disposed. This would make the code changes way less invasive. It should also be a separate pull request since it's quite unrelated to the rest (and can be used standalone). I'm also not convinced I see the advantage of having the first frame all the time.